modules/xfem/doc/content/modules/xfem/theory/theory.md

```diff
@@ -1,4 +1,127 @@
+
+Although the EFA is illustrated here using a 2D mesh and the applications shown in this paper are limited to 2D, this algorithm is readily applicable to 3D models. When there are distinctions between the 2D and 3D cases, those have been described in the above description of the algorithm, and these generally are related to the use of higher dimensionality geometric primitives (e.g. faces instead of edges) for the 3D case.  Also, this is not directly an issue with the EFA, but when extending this approach to 3D, there are other challenges such as continuity of the cutting planes between adjacent cracked elements [!cite](Duan2009) that must also be addressed.
+
+Finally, it should be noted that in the EFA as described above, if two child elements cut by the same crack have an edge that their parents share in common, but that edge contains no physical material, the nodes on that edge are not merged. This results in an extra phantom node, and hence an extra degree of freedom for every solution variable at that node. This behavior differs from the original XFEM, which would only add one degree of freedom to that node per field variable component. This scenario is shown on the patch of elements shown in [virtual_edge_merge].  The MOOSE implementation of this algorithm provides an option to merge these edges to recover the original XFEM behavior, and this is used in the present work.
+
+!media xfem/virtual_edge_merge.png
+       id=virtual_edge_merge
+       style=width:50%;margin-right:auto;margin-left:auto;
+       caption=Scenario in which the unmodified element fragment algorithm results in extra degrees of freedom. Without special treatment, there are two coincident phantom nodes at the indicated location because the neighboring edges share no physical material.
+
+## Partial Element Integration Algorithm
+
+Aside from the mesh cutting algorithm, the other major modification that must be made to the finite element code for the phantom node algorithm is that quadrature rules that are used to perform volume integrals, such as for computing the integral of the divergence of the stress tensor to compute an element's contribution to the residual, are adjusted to integrate over the physical portions of the cut elements. This is commonly done by forming a set of triangular (2D) or tetrahedral (3D) sub-elements that cover the entire physical portion of the split element's domain. These sub-elements are used only for integration, and new integration points are introduced at the locations of those sub-elements' integration points.
```

Comment:
Should we have references for element triangularization integration methods?

Reply:
I think we cited Dolbow's paper before and that one has the element triangularization.